tensorboard/plugins/pr_curve/tf_pr_curve_dashboard/tf-pr-curve-steps-selector.html

```diff
@@ -0,0 +1,194 @@
+      <div class="run-widget">
+        <div class="run-display-container">
+          <div class="run-color-box" style="background:[[_computeColorForRun(run)]];"></div>
+          <div class="run-text">
+            [[run]]
+          </div>
+        </div>
+        <div class="step-display-container">
+          [[_computeTimeTextForRun(runToAvailableTimeEntries, _runToStepIndex, run, timeDisplayType)]]
+        </div>
+        <paper-slider
+            data-run$="[[run]]"
+            step="1"
+            type="number"
+            min=0
+            max="[[_computeMaxStepIndexForRun(runToAvailableTimeEntries, run)]]"
+            ></paper-slider>
+      </div>
+    </template>
+    <style>
+      .run-widget {
+        margin: 10px 0 0 0;
+      }
+      paper-slider {
+        margin: -8px 0 0 -15px;
+        width: 100%;
+      }
+      .step-display-container {
+        font-size: 0.9em;
+        margin: 0 15px 0 0;
+      }
+      .run-text {
+        display: inline-block;
+      }
+      .run-color-box {
+        width: 12px;
+        height: 12px;
+        border-radius: 3px;
+        display: inline-block;
+      }
+    </style>
+  </template>
+  <script>
+  import {runsColorScale} from "../tf-color-scale/colorScale.js";
+
+  Polymer({
+    is: "tf-pr-curve-steps-selector",
+    properties: {
+      // A list of runs to create sliders for.
+      runs: Array,
+      // A mapping between run and list of steps to use for the slider for that
+      // run. This component indexes into this object to retrieve the steps.
+      runToAvailableTimeEntries: Object,
+      // The object mapping run to step. This object is updated by this
+      // component when the selected step for the run changes, so this property
+      // should be upward bound.
+      runToStep: {
+        type: Object,
+        notify: true,
+        computed: "_computeRunToStep(runToAvailableTimeEntries, _runToStepIndex)",
+      },
+      timeDisplayType: String,
+      // A mapping between run and slider index.
+      _runToStepIndex: Object,
+    },
+    listeners: {
+      "value-change": "_sliderValueChanged",
+      "immediate-value-change": "_sliderValueChanged",
+    },
+    observers: [
+      "_runsChanged(runs, _runToStepIndex)",
+      "_runToAvailableTimeEntriesChanged(runToAvailableTimeEntries)",
+    ],
+    ready() {
+      // Initialize the sliders so they point to the most recent value.
+      this._updateSliders(this._runToStepIndex);
+    },
+    _runsChanged(runs) {
+      // When selected runs change, make sure any new sliders point to the
+      // correct step. We use a setTimeout to add this operation to the end of
+      // the JS event queue so that it happens after the dom-repeat renders. We
+      // do not make this handler run when _runToStepIndex is updated because
+      // asynchronously changing _runToStepIndex could create a rendering loop
+      // and jittery sliders.
+      setTimeout(() => {
```

Comment:
Can this be replaced with `this.async(() => { … })`?

Reply:
Yes! Done.